rasa/core/policies/ted_policy.py

```diff
@@ -0,0 +1,642 @@
+            self.config[EVAL_NUM_EPOCHS] = self.config[EPOCHS]
+        elif self.config[EVAL_NUM_EPOCHS] < 1:
+            raise ValueError(
+                f"'{EVAL_NUM_EXAMPLES}' is set to '{self.config[EVAL_NUM_EPOCHS]}'. "
+                f"Only values > 1 are allowed for this configuration value."
+            )
+
+    # data helpers
+    # noinspection PyPep8Naming
+    @staticmethod
+    def _label_ids_for_Y(data_Y: np.ndarray) -> np.ndarray:
+        """Prepare Y data for training: extract label_ids."""
+
+        return data_Y.argmax(axis=-1)
+
+    # noinspection PyPep8Naming
+    def _label_features_for_Y(self, label_ids: np.ndarray) -> np.ndarray:
+        """Prepare Y data for training: features for label_ids."""
+
+        # full dialogue featurizer is used
+        if len(label_ids.shape) == 2:
+            return np.stack(
+                [
+                    np.stack(
+                        [
+                            self._label_data.get("label_features")[0][label_idx]
+                            for label_idx in seq_label_ids
+                        ]
+                    )
+                    for seq_label_ids in label_ids
+                ]
+            )
+
+        # max history featurizer is used
+        return np.stack(
+            [
+                self._label_data.get("label_features")[0][label_idx]
+                for label_idx in label_ids
+            ]
+        )
+
+    # noinspection PyPep8Naming
+    def _create_model_data(
+        self, data_X: np.ndarray, data_Y: Optional[np.ndarray] = None
+    ) -> RasaModelData:
+        """Combine all model related data into RasaModelData."""
+
+        label_ids = np.array([])
+        Y = np.array([])
+
+        if data_Y is not None:
+            label_ids = self._label_ids_for_Y(data_Y)
+            Y = self._label_features_for_Y(label_ids)
+            # explicitly add last dimension to label_ids
+            # to track correctly dynamic sequences
+            label_ids = np.expand_dims(label_ids, -1)
+
+        model_data = RasaModelData(label_key="label_ids")
+        model_data.add_features("dialogue_features", [data_X])
+        model_data.add_features("label_features", [Y])
+        model_data.add_features("label_ids", [label_ids])
+
+        return model_data
+
+    def _create_label_data(self, domain: Domain) -> RasaModelData:
+        # encode all label_ids with policies' featurizer
+        state_featurizer = self.featurizer.state_featurizer
+        all_labels = state_featurizer.create_encoded_all_actions(domain)
+        all_labels = all_labels.astype(np.float32)
+
+        label_data = RasaModelData()
+        label_data.add_features("label_features", [all_labels])
+        return label_data
+
+    # training methods
+    def train(
+        self,
+        training_trackers: List[DialogueStateTracker],
+        domain: Domain,
+        **kwargs: Any,
+    ) -> None:
+        """Train the policy on given training trackers."""
```

Comment:
please use a full docstring (also describing the parameters)

Reply:
also ok only to do it in the superclass imo, but then I would get rid of this one

Reply:
superclass has google docstring, I'd keep this one here though